drivers/rtc/rtc_ll_stm32.c

```diff
@@ -147,28 +265,44 @@ static int rtc_stm32_init(const struct device *dev)
 
 	k_mutex_init(&data->lock);
 
-	/* Enable Backup access */
-#if defined(PWR_CR_DBP) || defined(PWR_CR1_DBP) || defined(PWR_DBPCR_DBP) || defined(PWR_DBPR_DBP)
+#if RTC_STM32_BACKUP_DOMAIN_WRITE_PROTECTION
 	LL_PWR_EnableBkUpAccess();
-#endif /* PWR_CR_DBP || PWR_CR1_DBP || PWR_DBPR_DBP */
+#endif /* RTC_STM32_BACKUP_DOMAIN_WRITE_PROTECTION */
```

Comment:
Not directly linked to alarm support, please make it a dedicated commit

Reply:
Could be made clean in another PR.

Reply:
I prefer to get it now to be sure it is done ;)